src/app/vertextool/qgsvertextool.cpp

```diff
@@ -2228,37 +2228,53 @@ void QgsVertexTool::moveVertex( const QgsPointXY &mapPoint, const QgsPointLocato
     addExtraSegmentsToEdits( edits, mapPoint, dragLayer, layerPoint );
   }
 
+  VertexEdits beforeEdits;
+  const auto editKeys = edits.keys();
+  for ( QgsVectorLayer *layer : editKeys )
+  {
+    const auto editFid = edits[layer].keys();
```

Comment:
Could you iterate on key, value to avoid [container-anti-pattern](https://github.com/KDE/clazy/blob/master/docs/checks/README-container-anti-pattern.md)

Strange that clang-tidy doesn't scream on this one, because maybe of the `auto` ?